api/v1beta1/moduleimagesconfig_types.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ImageState string
+
+const (
+	// ImageExists means that image exists in the specified repo
+	ImageExists ImageState = "Exists"
+	// ImageNotExists means that image does not exist in the specified repo
+	ImageNotExists ImageState = "NotExists"
+)
+
+// ModuleImagesConfigSpec describes the images of the Module whose status needs to be verified
+// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+// +kubebuilder:validation:Required
+type ModuleImagesConfigSpec struct {
+	Images []string `json:"images"`
```

Comment:
How about we call this field `Items`? Also, it is not a slice of strings but a slice of structs (you are missing the `generation` field for each `Item`.

Reply:
The inner object can be `ModuleImageSpec` to fit with the following `ModuleImageStatus` later on.

Reply:
We can also go with `ImageItem`

Reply:
I think that `ModuleImageConfig` is too similar to `ModuleImagesConfig`.